Math-Interactions-Overview.md

```diff
@@ -3,26 +3,30 @@ There are currently three interactions that can be used for math-related questio
-- `guppy-initialization-service.ts`: Service for initializing a guppy instance. On the creator's dashboard, there could be multiple guppy instances present at any given point. This service manages all of these instances and helps in initializing a new instance and finding the currently active instance.
+- `guppy-initialization-service.ts`: Service for initializing a guppy instance. On the exploration editor page, there could be multiple guppy instances present at any given point. This service manages all of these instances and helps in initializing a new instance and finding the currently active instance.
 - `guppy-configuration-service.ts`: Service for handling the global config that will be applied to all guppy instances. 
 - `math-interaction-service.ts`: The core service for handling most of the math interaction-related functionalities on the frontend. This service handles:
   - All frontend validations using nerdamer.
   - Modifying the user input from guppy and transforming it such that it is compatible with nerdamer and our backend parser schema.
   - Generating error messages for various invalid scenarios.
 
-For the custom on-screen keyboard, the functionality is handled by the component present in `core/templates/components/on-screen-keyboard`. This does not extend the custom OSK provided by guppy since customization on that is quite limited. This contains normal buttons that call the guppy API functions to insert characters onto the currently active guppy instance.
+For the custom on-screen keyboard, the functionality is handled by the component present in `core/templates/components/on-screen-keyboard`. This does not extend the custom OSK provided by guppy since customization on that is quite limited. The custom on-screen keyboard contains normal buttons that call the guppy API functions to insert characters onto the currently active guppy instance.
 
 ### Backend
 There are some custom objects (e.g. `MathEquation`, `AlgebraicIdentifier`, etc.)  present in `extensions/objects/models/objects.py` that help in type checks for the objects used in these interactions. The validators (e.g. `is_valid_math_expression`, `is_valid_algebraic_expression`, etc.) for these objects are present in `core/schema_utils.py`. 
 
-For backend validations of the data stored for these interactions, we have a custom expression parser in `core/domain/expression_parser.py`. **Note that these need to be in sync with the behavior of nerdamer's validations.**
+For backend validations of the data stored for these interactions, we have a custom expression parser in `core/domain/expression_parser.py`. **Note that these need to be in sync with the behavior of nerdamer's validations.** To check nerdamer's behaviour, the source code can be checked [here](https://github.com/jiggzson/nerdamer). To ensure sync, an exhaustive list of test cases (refer to the original proposal) are run on both frontend and backend validators, so these can be updated if and when required.
```

Comment:
Please link to the "original proposal", or better still, the specific part of the document that details the test cases. Explain, also, where these test cases are located in the Oppia codebase.

Reply:
I did link the proposal. Mentioned the section here. Also added location of test cases.